dowhy/gcm/unit_change.py

```diff
@@ -0,0 +1,140 @@
+"""This module provides the APIs for attributing the change in the output value of a deterministic mechanism for a statistical unit.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from sklearn.utils.validation import check_is_fitted
+from typing_extensions import Protocol
+
+from dowhy.gcm.shapley import ShapleyConfig, estimate_shapley_values
+
+
+class CanEvaluateInput(Protocol):
+    """This protocol defines a trait for classes that can evaluate a given input."""
+
+    @abstractmethod
+    def evaluate(self, input: np.ndarray) -> np.ndarray:
+        pass
+
+
+class HasCoefficients(Protocol):
+    """This protocol defines a trait for classes that have coefficients."""
+
+    @property
+    @abstractmethod
+    def coefficients(self) -> np.ndarray:
+        pass
+
+
+class Mechanism(CanEvaluateInput, ABC):
+    pass
+
+
+class LinearMechanism(Mechanism, HasCoefficients):
+    pass
+
+
+class SklearnMechanism(Mechanism):
+    """Represents any Sklearn prediction model."""
+
+    def __init__(self, fitted_sklearn_model) -> None:
```

Comment:
Can we use type hints for `fitted_sklearn_model`? (if necessary, feel free to define another Protocol)